src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -1047,3 +1052,44 @@ void ControllerScriptInterfaceLegacy::softStart(int deck, bool activate, double
     // activate the ramping in scratchProcess()
     m_ramp[deck] = true;
 }
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const ControllerScriptInterfaceLegacy::WellKnownCharsets targetCharset,
+        const QString& value) {
+    switch (targetCharset) {
+    case WellKnownCharsets::LATIN_1:
+    case WellKnownCharsets::ISO_8859_1:
+        return this->convertCharset(QStringLiteral("ISO-8859-1"), value);
+    case WellKnownCharsets::LATIN_9:
+    case WellKnownCharsets::ISO_8859_15:
+        return this->convertCharset(QStringLiteral("ISO-8859-15"), value);
+    case WellKnownCharsets::UCS_2:
+    case WellKnownCharsets::ISO_10646_UCS_2:
+        return this->convertCharset(QStringLiteral("ISO-10646-UCS-2"), value);
+    default:
+        m_pScriptEngineLegacy->throwJSError(QStringLiteral("Unknown charset specified"));
+        return QByteArray();
+    }
+}
+
+QByteArray ControllerScriptInterfaceLegacy::convertCharset(
+        const QString& targetCharset, const QString& value) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 4, 0)
+    auto* pCodec = QTextCodec::codecForName(targetCharset.toUtf8());
+    if (!pCodec) {
+        return QByteArray();
```

Comment:
I think we also want logoutput here and/or throw an error. wdyt?

Reply:
Like:

```c++
VERIFY_OR_DEBUG_ASSERT(…) {

}
jsEngine()->throwError()
```
?

Reply:
How exactly does it fail currently? I'd need to see some code to make a proper judgement.

Reply:
yeah, thats why `logMessagesExpected` is quite a hack...